### ISSUE TITLE
`WARNING  Unable to load module ... for options validation`

### ISSUE THREAD

**Issue by Vanav** [ansible/ansible-lint#2828]
`ansible-lint` displays warnings:
```
WARNING: PATH altered to include /opt/pipx/venvs/ansible-lint/bin :: This is usually a sign of broken local setup, which can cause unexpected behaviors.
...
WARNING  Unable to load module snap at roles/base/tasks/main.yml:32 for options validation
WARNING  Unable to load module timezone at roles/base/tasks/main.yml:60 for options validation
WARNING  Unable to load module zabbix_group at roles/zabbix/tasks/main.yml:165 for options validation
WARNING  Unable to load module zabbix_host at roles/zabbix/tasks/main.yml:172 for options validation
WARNING  Unable to load module zabbix_hostmacro at roles/zabbix/tasks/main.yml:209 for options validation
WARNING  Unable to load module synchronize at roles/deploy/tasks/update-code/rsync_direct.yml:13 for options validation
WARNING  Unable to load module community.general.snap at roles/nginx/tasks/main.yml:60 for options validation
WARNING  Unable to load module postgresql_user at roles/postgres-env/tasks/main.yml:5 for options validation
WARNING  Unable to load module postgresql_db at roles/postgres-env/tasks/main.yml:12 for options validation
WARNING  Unable to load module postgresql_ext at roles/postgres-env/tasks/main.yml:19 for options validation
WARNING  Unable to load module git_config at roles/base/tasks/Debian.yml:291 for options validation
WARNING  Unable to load module locale_gen at roles/base/tasks/Debian.yml:370 for options validation
WARNING  Unable to load module authorized_key at roles/unix-users/tasks/main.yml:110 for options validation
```
I have installed ansible and ansible-lint in pipx environments, as recommended:
```
# ansible-lint --version
WARNING: PATH altered to include /opt/pipx/venvs/ansible-lint/bin :: This is usually a sign of broken local setup, which can cause unexpected behaviors.
ansible-lint 6.10.0 using ansible 2.14.0

# pipx list
venvs are in /opt/pipx/venvs
apps are exposed on your $PATH at /opt/pipx/bin
   package ansible 7.1.0, installed using Python 3.10.6
    - ansible
    - ansible-community
    - ansible-config
    - ansible-connection
    - ansible-console
    - ansible-doc
    - ansible-galaxy
    - ansible-inventory
    - ansible-playbook
    - ansible-pull
    - ansible-test
    - ansible-vault
   package ansible-lint 6.10.0, installed using Python 3.10.6
    - ansible-lint
```
Playbook has no issues. I've tried to use FQCN for modules, but it doesn't change anything: same warnings, but with FQCN paths.

**Comment by netsandbox** [comment]
I'm also see this warnings, but other than @Vanav report, I actually activated the venv.

```shell
$ ansible-lint
WARNING: ansible-lint is no longer tested under Python 3.8 and will soon require 3.9. Do not report bugs for this version.
WARNING  Unable to load module ansible.windows.win_copy at playbooks/ssl_cert_windows.yaml:25 for options validation
WARNING  Unable to load module ansible.windows.win_certificate_store at playbooks/ssl_cert_windows.yaml:31 for options validation
WARNING  Unable to load module ansible.windows.win_file at playbooks/ssl_cert_windows.yaml:37 for options validation
WARNING  Unable to load module community.windows.win_disk_facts at playbooks/facts.yaml:23 for options validation
```

```shell
$ ansible-galaxy collection list

# /home/ansible/collections/ansible_collections
Collection            Version
--------------------- -------
ansible.posix         1.4.0  
ansible.windows       1.12.0 
community.general     6.0.1  
community.hashi_vault 4.0.0  
community.vmware      2.10.2 
community.windows     1.11.1 
```

**Comment by nickkeyzer** [comment]
Possibly related but I'm seeing the same symptoms as described here for all Windows modules: https://github.com/ansible/ansible-lint/issues/2831

**Comment by hemberger** [comment]
I don't see this issue when I run ansible-lint locally (v6.10.0), but I do see it in my Actions workflow. I've reported it here: https://github.com/ansible/ansible-lint-action/issues/99#issuecomment-1365487875.

**Comment by Schachte** [comment]
This is affecting me as well. 

### requirements.yaml

```
---
collections:
  - name: community.docker
    source: https://galaxy.ansible.com
```

In my action, I pre-install this.

```
name: Ansible Deployment

on:
  workflow_dispatch:

jobs:
  linting:
    name: List Ansible Files
    runs-on: ubuntu-22.04

    steps:
    - name: Checkout Sources
      uses: actions/checkout@v2

    - name: Install Ansible
      run: sudo apt-get install -y ansible

    - name: Install Ansible-Galaxy Requirements
      run: ansible-galaxy install -r requirements.yaml

    - name: List Installed Ansible-Galaxy Requirements
      run: ansible-galaxy collection list

    - name: Linting
      uses: ansible/ansible-lint-action@dd225474d15635904348151b9e694472ce0f62a8
      with:
        path: ansible/
```

I see the output of the `collection list` showing exactly what I installed, but the Github Action fails to resolve with the following:


```
WARNING  Unable to load module community.docker.docker_compose at ansible/playbooks/roles/nginx/tasks/nginx.yaml:2 for options validation
[10](https://github.com/Schachte/homelab/actions/runs/3813289308/jobs/6486951382#step:7:11)
WARNING  Unable to load module community.docker.docker_compose at ansible/playbooks/roles/nginx/tasks/nginx.yaml:7 for options validation
[11](https://github.com/Schachte/homelab/actions/runs/3813289308/jobs/6486951382#step:7:12)
WARNING  Unable to load module community.docker.docker_compose at ansible/playbooks/roles/nginx/tasks/nginx.yaml:12 for options validation
[12](https://github.com/Schachte/homelab/actions/runs/3813289308/jobs/6486951382#step:7:13)
WARNING  Unable to resolve FQCN for module community.docker.docker_compose
```

**Comment by ssbarnea** [comment]
I will be converting this to a discussion as there is no specific bug identified. I will fine a bug once we identify a clear bug. When it comes to use of GHA action, there are several things people need to remember because that is running inside a containers, so doing any preparatory work is unlikely to help in a any way.
